sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -494,4 +494,45 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
         spark.table("t").write.insertInto(tableName)
       }
   }
+
+  private def dropTables(tableNames: String*): Unit = {
+    tableNames.foreach { name =>
+      sql(s"DROP TABLE IF EXISTS $name")
+    }
+  }
+
+  test(
+    """SPARK-20594: The staging directory should be appended with ".hive-staging"
+      |to avoid being deleted if we set hive.exec.stagingdir under the table directory
+      |without start with "."""".stripMargin) {
+
+    dropTables("test_table", "test_table1")
+
+    sql("CREATE TABLE test_table (key int, value string)")
+
+    // Add some data.
+    testData.write.mode(SaveMode.Append).insertInto("test_table")
```

Comment:
no, as i tested we must create table rather than simplify the above two lines by `spark.range(1).write.saveAsTable("test_table")`.